lckernel/cad/geometry/geobezier.cpp

```diff
@@ -0,0 +1,83 @@
+#include "geobezier.h"
+
+using namespace lc;
+using namespace geo;
+
+Bezier::Bezier(const Coordinate& point_a, const Coordinate& point_b, const Coordinate& point_c)
+    : _pointA(point_a), _pointB(point_b), _pointC(point_c) {
+}
+
+Bezier::Bezier(const Bezier &bez) :
+    _pointA(bez._pointA), _pointB(bez._pointB),
+    _pointC(bez._pointC) {
+
+}
+
+const Area Bezier::boundingBox() const {
+    /*
+     * TODO BOUNDING BOX
+     * GSOC project
+     */
+
+    return Area();
+}
+
+Coordinate Bezier::nearestPointOnPath(const Coordinate& coord) const {
+    /*
+     * TODO
+     */
+    return Coordinate();
+}
+
+Coordinate Bezier::nearestPointOnEntity(const Coordinate& coord) const {
+    /*
+     * TODO
+     */
+    return Coordinate();
+}
+
+Coordinate Bezier::CasteljauAt(std::vector<Coordinate> points, float t) const {
+    if(points.size() == 1) return points[0];
+    else {
+        int size_ = points.size() - 1   ;
+        std::vector<Coordinate> new_vec;
+        for(int i = 0; i < size_; i++) {
+            double _x = (1-t) * points[i].x() + t * points[i+1].x();
+            double _y = (1-t) * points[i].y() + t * points[i+1].y();
+            new_vec.push_back(Coordinate(_x, _y));
+        }
+        return CasteljauAt(new_vec, t);
```

Comment:
trying to make it compact here. Should be easier for maintenance

```C++
Coordinate Bezier::CasteljauAt(std::vector<Coordinate> points, float t) const {
 if(points.size() == 1) return points[0];
std::vector<Coordinate> new_vec;
for (size_t i = 1; i < points.size(); i++)
     new_vec.push_back( points[i-1] * (1 - t) + points[i] * t);
return new_vec;
}
```

Anyway, most time, we should be using matrix based algorithms for efficiency.



}
```